src/resources/js/views/datepicker.js

```diff
@@ -0,0 +1,289 @@
+	 */
+	obj.handleChangeDate = function( event ) {
+		/**
+		 * @todo: handle week view case here.
+		 */
+		var $container = event.data.container;
+		var date = event.date.getDate();
+		var month = event.date.getMonth() + 1;
+		var year = event.date.getFullYear();
+
+		var paddedDate = obj.padNumber( date );
+		var paddedMonth = obj.padNumber( month );
+
+		var viewData = {
+			[ 'tribe-bar-date' ]: [ year, paddedMonth, paddedDate ].join( '-' ),
+		};
+
+		obj.request( viewData, $container );
+	};
+
+	/**
+	 * Handle datepicker changeMonth event
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object for 'changeMonth' event
+	 *
+	 * @return {void}
+	 */
+	obj.handleChangeMonth = function( event ) {
+		var $container = event.data.container;
+		var month = event.date.getMonth() + 1;
+		var year = event.date.getFullYear();
+
+		var paddedMonth = obj.padNumber( month );
+
+		var viewData = {
+			[ 'tribe-bar-date' ]: [ year, paddedMonth ].join( '-' ),
+		};
+
+		obj.request( viewData, $container );
+	};
+
+	/**
+	 * Handle datepicker hide event
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object for 'hide' event
+	 *
+	 * @return {void}
+	 */
+	obj.handleHide = function( event ) {
+		event.data.datepickerButton.removeClass( obj.selectors.buttonOpenClass.className() );
+	};
+
+	/**
+	 * Show datepicker on datepicker button click
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object for 'click' event
+	 *
+	 * @return {void}
+	 */
+	obj.showDatepicker = function( event ) {
+		var $input = $( event.data.input );
+		var $datepickerButton = $( event.data.target );
+
+		$datepickerButton.toggleClass( obj.selectors.buttonOpenClass.className() );
+		$input
+			.focus()
+			.bootstrapDatepicker( 'show' );
+	};
+
+	/**
+	 * Initialize datepicker JS
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object for 'afterSetup.tribeEvents' event
+	 * @param {integer} index jQuery.each index param from 'afterSetup.tribeEvents' event
+	 * @param {jQuery} $container jQuery object of view container
+	 * @param {object} data data object passed from 'afterSetup.tribeEvents' event
+	 *
+	 * @return {void}
+	 */
+	obj.init = function( event, index, $container, data ) {
+		var $input = $container.find( obj.selectors.input );
+		var $datepickerButton = $container.find( obj.selectors.button );
+		var viewSlug = data.slug;
+		var isMonthView = 'month' === viewSlug;
+		var minViewMode = isMonthView ? 'year' : 'month';
+		var changeEvent = isMonthView ? 'changeMonth' : 'changeDate';
+		var changeHandler = isMonthView ? obj.handleChangeMonth : obj.handleChangeDate;
+
+		var tribeL10nDatatables = window.tribe_l10n_datatables || {};
+		var datepickerI18n = tribeL10nDatatables.datepicker || {};
+		var nextText = datepickerI18n.nextText || 'Next';
+		var prevText = datepickerI18n.prevText || 'Prev';
+
+		$input
+			.bootstrapDatepicker( {
+				container: $input.closest( obj.selectors.datepickerContainer ),
+				format: 'yyyy-mm-dd',
```

Comment:
We should be printing out the date format selected on the settings in a `data-js` and then grabbing the value from there. Maybe not a stopper now, but we need to implement that.

Reply:
Yes, you're right. I forgot to leave a todo here. If anything, we'll likely pass the format from the BE to the window object and use that value here. I'll add a todo to tackle as we hook up the events bar.